lib/pacemaker/pcmk_sched_instances.c

```diff
@@ -632,9 +661,7 @@ find_compatible_child_by_node(const pe_resource_t *local_child,
     }
 
     crm_trace("Can't pair %s with %s", local_child->id, rsc->id);
-    if(children != rsc->children) {
-        g_list_free(children);
-    }
+    free_instance_list(rsc, children);
```

Comment:
We might return early before we get here

Reply:
thanks